lib/evmone/baseline.cpp

```diff
@@ -67,20 +68,36 @@ CodeAnalysis analyze_legacy(bytes_view code)
     return {pad_code(code), code.size(), analyze_jumpdests(code)};
 }
 
-CodeAnalysis analyze_eof1(bytes_view eof_container, const EOF1Header& header)
+CodeAnalysis analyze_eof1(bytes_view container)
 {
-    const auto executable_code = eof_container.substr(header.code_begin(), header.code_size);
-    return {executable_code, analyze_jumpdests(executable_code)};
+    const auto header = read_valid_eof1_header(container);
+
+    // Extract all code sections as single buffer reference.
+    // TODO: It would be much easier if header had code_sections_offset and data_section_offset
+    //       with code_offsets[] being relative to code_sections_offset.
+    const auto code_sections_offset = header.code_offsets[0];
+    const auto code_sections_end = size_t{header.code_offsets.back()} + header.code_sizes.back();
+    const auto executable_code =
+        container.substr(code_sections_offset, code_sections_end - code_sections_offset);
+
+    // Offsets relative to first code section.
+    std::vector<uint16_t> relative_offsets;
+    relative_offsets.reserve(header.code_offsets.size());
+    for (const auto offset : header.code_offsets)
+        relative_offsets.push_back(offset - code_sections_offset);
+
+    // FIXME: Better way of getting EOF version.
+    auto analysis = CodeAnalysis{executable_code, {}, container[2]};
```

Comment:
Hard to decipher the meaning of `container[2]` without looking at the definition of `CodeAnalysis`. Would suggest:
```suggestion
    const auto eof_version = container[2];
    auto analysis = CodeAnalysis{executable_code, {}, eof_version};
```

Or passing down the version from `read_valid_eof1_header` or including it in `EOF1Header`.